pip/resolve.py

```diff
@@ -123,11 +123,26 @@ class Resolver(object):
     the requested operation without breaking the requirements of any package.
     """
 
-    def __init__(self, upgrade_strategy, finder):
+    _allowed_strategies = ["eager", "only-if-needed", "to-satisfy-only"]
```

Comment:
Minor nit, it'd be nice if this was a set instead of a list. The order doesn't matter and we're only testing for membership in it and a set is faster for that. For a list this small it doesn't really matter, but it's a good habit to get into either way.

Since we're Python 2.7+ you can just use set literals for it, ``{"eager", "only-if-needed", "to-satsify-only"}``.